Foundation/NSString.swift

```diff
@@ -1423,10 +1423,20 @@ extension NSMutableString {
 
         if let findResults = CFStringCreateArrayWithFindResults(kCFAllocatorSystemDefault, _cfObject, target._cfObject, CFRange(searchRange), options._cfValue(true)) {
             let numOccurrences = CFArrayGetCount(findResults)
+            let start = _storage.utf16.startIndex
```

Comment:
_storage should only be accessed in concrete methods, it may not always be there

Reply:
This might be tricky, without factory patterns which allow us to return a special NSMutableString subclass.